src/tmpfiles/tmpfiles.c

```diff
@@ -2003,6 +2005,182 @@ static int glob_item_recursively(Item *i, fdaction_t action) {
         return r;
 }
 
+static int rm_if_wrong_type_safe(
+                mode_t mode,
+                int parent_fd,
+                const struct stat *parent_st /* Only used if follow is true. */,
+                const char *name,
+                int flags) {
+        _cleanup_free_ char *parent_name = NULL;
+        bool follow_links = (flags & AT_SYMLINK_NOFOLLOW) == 0;
+        struct stat st;
+        int r;
+
+        assert(name);
+        assert((mode & ~S_IFMT) == 0);
+        assert(!follow_links || parent_st);
+        assert((flags & ~AT_SYMLINK_NOFOLLOW) == 0);
+
+        if (!filename_is_valid(name))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "\"%s\" is not a valid filename.", name);
+
+        if (fstatat(parent_fd, name, &st, flags) < 0) {
+                (void) fd_get_path(parent_fd, &parent_name);
```

Comment:
fd_get_path() might corrupt errno,  but you are using it below. Please save it via `r = -errno` or so, before you call fd_get_path(), and then use it in the log message